src/devices/cardin.c

```diff
@@ -92,40 +102,56 @@ static int cardin_callback(bitbuffer_t *bitbuffer) {
 				dip[8]='+';
 		}
 
-		fprintf(stdout, "                 123456789\n");
-		fprintf(stdout, "dipswitch      = %s\n\n",dip);
-
-		fprintf(stdout, "                 -->ON\n");
-		fprintf(stdout, "right button   = ");
+		// Jumpers for the right button
 		if((bb[0][2] & 3) == 3) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 --o\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "--o");
 		}
 		if((bb[0][2] & 9) == 9) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "o--");
 		}
 		if((bb[0][2] & 12) == 12) {
-			fprintf(stdout,                  "2 o-- (this is left button or two buttons on same channel)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "o-- (this is left button or two buttons on same channel)");
```

Comment:
great solution! On last nit-picking: `rbutton[((bb[0][2] & 15) / 3)-1]` will go out-of-bounds for bb[0][2] < 3 or >= 15. Maybe the intent is clearer with just 16 strings of:
```
char *rbutton[4] = { "?", "?", "?", "11R", "?", "?", "10R", "?", "?", "01R", "?", "?", "00L?", "?", "?", "?" };
```